uwuno/src/main/java/com/learning/uwuno/room.java

```diff
@@ -27,19 +36,34 @@ public String getUid() {
         return uid;
     }
 
-    public boolean addPlayer(player newPlayer) {
-        return playerList.add(newPlayer);
-    }
-
+    // Player Functions
     public ArrayList<player> getPlayers() {
         return playerList;
     }
 
     public player getPlayer(String pid) {
-        return playerList.stream().filter(t -> t.getUid().equals(pid)).findFirst().get();
+        return playerList.stream().filter(t -> t.getPid().equals(pid)).findFirst().get();
+    }
+
+    public void addPlayer(player newPlayer) {
+        playerList.add(newPlayer);
     }
 
     public void deletePlayer(String pid) {
-        playerList.removeIf(t -> t.getUid().equals(pid));
+        playerList.removeIf(t -> t.getPid().equals(pid));
+    }
+
+    // Deck Functions, do not return deck and handle everything here so as there's no way to control what is
+    // done with the reference of the deck
+    public ArrayList<card> drawCards(int numCards) {
```

Comment:
I feel like all these card handling also in gameServices, should be done through the player instead of the room same with drawHand in Deck. But if it works, then we should just leave it

Reply:
Player doesn't have access to the deck without using room as an interface atm, is there another way classes can be set up to access each other's variables?

Reply:
The only way I can think of is to pass deck as a parameter into player and these functions can be all moved there? Don't know how dumb that sounds tho.